scripts/setup/nxp/update_nxp_sdk.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+CHIP_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../..'))
+
+# Array listing all SDK and default targeted path for NXP SDK storage
+ALL_PLATFORM_SDK = [
+    {'plat_name': 'k32w0', 'plat_sdk_manifest_path': 'third_party/nxp/nxp_matter_support/github_sdk/k32w0'},
+]
+
+all_platform_sdk_list = list(map(lambda plat: plat["plat_name"], ALL_PLATFORM_SDK))
+
+
+def clean_sdk(sdk_target_location):
+    print("SDK will be cleaned all local modification(s) will be lost")
+    # Cleaning all local modifications
+    git_clean_command = "git reset --hard && git clean -xdf"
+    command = ['west', 'forall', '-c', git_clean_command, '-a']
+    subprocess.run(command, cwd=sdk_target_location, check=True)
+    # Retrying an update
```

Comment:
why is this a "retry"?  It would be nice for the code to explain why things are done - I can see it does an update_platform_sdk_version, but not as clear why.

What does `False` mean?

Reply:
I agree, it is not very clear and the location of the retry is not correct. I re-worked this part my latest commit and moved this "retry" mechanism in a better place.  Please let me know if you are ok.